src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Reflection/Extensions/NonPortable/CustomAttributeSearcher.cs

```diff
@@ -110,7 +110,7 @@ public virtual E GetParent(E e)
         //
         private IEnumerable<CustomAttributeData> GetMatchingCustomAttributesIterator(E element, Func<Type, bool> passesFilter, bool inherit)
         {
-            LowLevelList<CustomAttributeData> immediateResults = new LowLevelList<CustomAttributeData>();
+            ArrayBuilder<CustomAttributeData> immediateResults = default;
```

Comment:
Can this use `ListBuilder`? ListBuilder avoids extra allocations for list of size 1 that I expect to be a very common case in this method.

Also, could you please add `if (inherit)` in front of `immediateResults.Add(cad);` a few lines below?